File: lib/java/opentoken/src/test/java/com/truveta/opentoken/tokens/TokenRegistryTest.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */

package com.truveta.opentoken.tokens;

import com.truveta.opentoken.attributes.AttributeExpression;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;

class TokenRegistryTest {

    @Test
    void testLoadAllTokensReturnsNonEmptyMap() {
        Map<String, List<AttributeExpression>> tokens = TokenRegistry.loadAllTokens();

        assertFalse(tokens.isEmpty(), "Tokens map should not be empty");

        String[] expectedTokens = { "T1", "T2", "T3", "T4", "T5" };

        for (String tokenId : expectedTokens) {
            assertTrue(tokens.containsKey(tokenId), "Tokens map should contain " + tokenId);

            List<AttributeExpression> definitions = tokens.get(tokenId);
            assertNotNull(definitions, "Definitions for " + tokenId + " should not be null");
            assertFalse(definitions.isEmpty(), "Definitions for " + tokenId + " should not be empty");
        }
    }
}

File: lib/python/opentoken/src/test/opentoken/tokens/token_registry_test.py
"""
Copyright (c) Truveta. All rights reserved.
"""
import pytest
from opentoken.tokens.token_registry import TokenRegistry

def test_load_all_tokens_returns_non_empty_dict():
    tokens = TokenRegistry.load_all_tokens()
    assert tokens, "Tokens dict should not be empty"

    expected_tokens = ["T1", "T2", "T3", "T4", "T5"]
    for token_id in expected_tokens:
        assert token_id in tokens, f"Tokens dict should contain {token_id}"
        definitions = tokens[token_id]
        assert definitions is not None, f"Definitions for {token_id} should not be None"
        assert definitions, f"Definitions for {token_id} should not be empty"
